db/document.go

```diff
@@ -52,22 +53,24 @@ type AttachmentsMeta map[string]interface{} // AttachmentsMeta metadata as inclu
-	Expiry          *time.Time          `json:"exp,omitempty"`                     // Document expiry.  Information only - actual expiry/delete handling is done by bucket storage.  Needs to be pointer for omitempty to work (see https://github.com/golang/go/issues/4357)
-	Cas             string              `json:"cas"`                               // String representation of a cas value, populated via macro expansion
-	Crc32c          string              `json:"value_crc32c"`                      // String representation of crc32c hash of doc body, populated via macro expansion
-	Crc32cUserXattr string              `json:"user_xattr_value_crc32c,omitempty"` // String representation of crc32c hash of user xattr
-	TombstonedAt    int64               `json:"tombstoned_at,omitempty"`           // Time the document was tombstoned.  Used for view compaction
-	Attachments     AttachmentsMeta     `json:"attachments,omitempty"`
+	CurrentRev      string                            `json:"rev"`
+	NewestRev       string                            `json:"new_rev,omitempty"` // Newest rev, if different from CurrentRev
+	Flags           uint8                             `json:"flags,omitempty"`
+	Sequence        uint64                            `json:"sequence,omitempty"`
+	UnusedSequences []uint64                          `json:"unused_sequences,omitempty"` // unused sequences due to update conflicts/CAS retry
+	RecentSequences []uint64                          `json:"recent_sequences,omitempty"` // recent sequences for this doc - used in server dedup handling
+	History         RevTree                           `json:"history"`
+	Channels        channels.ChannelMap               `json:"channels,omitempty"`
+	Access          UserAccessMap                     `json:"access,omitempty"`
+	RoleAccess      UserAccessMap                     `json:"role_access,omitempty"`
+	Expiry          *time.Time                        `json:"exp,omitempty"`                     // Document expiry.  Information only - actual expiry/delete handling is done by bucket storage.  Needs to be pointer for omitempty to work (see https://github.com/golang/go/issues/4357)
+	Cas             string                            `json:"cas"`                               // String representation of a cas value, populated via macro expansion
+	Crc32c          string                            `json:"value_crc32c"`                      // String representation of crc32c hash of doc body, populated via macro expansion
+	Crc32cUserXattr string                            `json:"user_xattr_value_crc32c,omitempty"` // String representation of crc32c hash of user xattr
+	TombstonedAt    int64                             `json:"tombstoned_at,omitempty"`           // Time the document was tombstoned.  Used for view compaction
+	Attachments     AttachmentsMeta                   `json:"attachments,omitempty"`
+	ChanNames       []string                          `json:"chan_names"`
```

Comment:
Are these just active channels, or do they also include previous channels?  We might want to think a bit more about how these might be indexed and queried in future to determine whether to include both active and previous channels in a single set, or to separate them.  Having to cross-reference against another array to differentiate might cause issues.  An alternative might be to store as ["+chan1", "-chan2"] so that they are indexed together, but that would mean two scans at query time to pick up both cases.

Reply:
ChanNames maps to ChannelHistory so would contain previous channels too. Like index 1 of chan names is associated with additions and removals of ChannelHistory in index 1

Reply:
You're right, need to discuss this I think.